[caffe2/torchgen] Fix codegen non-determinism

andrewjcg opened the pull request:
Fix several cases of leaking set-iteration-order to generated
sources, causing non-determinism in generated code.

albanD reviewed (commented):
The change sounds ok. Is there any way we could test this to ensure we don't regress going forward?

andrewjcg commented:
There's some strategies I've seen to detect this (e.g. running script multiple times with `PYTHONHASHSEED=random`).  Internally, we log invocations and their outputs, so we can use this data to detect these cases.

andrewjcg committed:
[caffe2/torchgen] Fix codegen non-determinism (#100986)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/100986

Fix several cases of leaking set-iteration-order to generated
sources, causing non-determinism in generated code.
